src/test/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapperTest.java

```diff
@@ -0,0 +1,421 @@
+
+        String[] expected = {
+                "-f", "/path/to/environment.yaml",
+                commandName,
+                PublishLambdaCommand.LAMBDA_NAME_LONG_ARG, "WAF",
+                PublishLambdaCommand.ARTIFACT_URL_LONG_ARG, "https://github.com/Nike-Inc/cerberus-cloudfront-lambda/releases/download/v1.1.0/cerberus-cloudfront-lambda.jar"
+        };
+
+        String[] actual = EnvironmentConfigToArgsMapper.getArgs(environmentConfig, userInput);
+
+        assertArgsAreEqual(expected, actual, commandName);
+    }
+
+    private void assertArgsAreEqual(String[] expected, String[] actual, String commandName) {
+        // the args up to and including the command should be the same
+        int commandIndex = -1;
+        for (int i = 0; i < expected.length; i++) {
+            String e = expected[i];
+            String a = actual[i];
+            assertEquals(e, a);
+            if (e.equals(commandName)) {
+                commandIndex = i;
+                break;
+            }
+        }
+
+        // everything after the command name is not guaranteed to be in the same order
+        // so we need to ensure that the pairs are in expected are also in the actual, as well as the single arg switches
+        // we will do an n^2 brute force check
+        for (int i = commandIndex + 1; i < expected.length; i++) {
+            String eKey = expected[i];
+
+            boolean isPairE = false;
+            String eVal = "";
+            if (i < expected.length - 1) {
+                eVal = expected[i+1];
+                if (! StringUtils.startsWith(eVal, "-")) {
+                    isPairE = true;
+                }
+            }
+
+            if (isPairE) {
+                i++;
+            }
+
+            boolean found = false;
+            for (int n = commandIndex + 1; n < actual.length; n++) {
+                String aKey = actual[n];
+
+                boolean isPairA = false;
+                String aVal = "";
+                if (n < actual.length - 1) {
+                    aVal = actual[n+1];
+                    if (! StringUtils.startsWith(aVal, "-")) {
+                        isPairA = true;
+                    }
+                }
+
+                if (isPairA) {
+                    n++;
+                }
+
+                if (isPairE && isPairA && eKey.equals(aKey) && eVal.equals(aVal)) {
+                    found = true;
+                    break;
+                } else if (! isPairE && ! isPairA && eKey.equals(aKey)) {
+                    found = true;
+                    break;
+                }
+            }
+            if (! found) {
+                fail(String.format("Failed to find [ %s %s ] in actual args \nactual args:   %s\nexpected args: %s\nactual size: %s\nexpected size: %s", eKey, eVal, String.join(" ", actual), String.join(" ", expected), actual.length, expected.length));

```

Comment:
- [ ] should have different print statement for when not searching for a k,v pair and just a flag.